src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -740,6 +740,36 @@ namespace smt::noodler {
         auto [input_vars_automata, input_vars_divisions] = solving_state.get_automata_and_division_of_concatenation(input_vars, true);
         SASSERT(input_vars_automata.size() == input_vars_divisions.size());
 
+        // In the case that input side leads to one automaton where we have only non-length vars, we can apply this automaton directly to the transducer.
+        // So if we have
+        //   output_vars = T(input_vars),
+        // we take the language of applying automaton from input_vars on T and create a new fresh_var with this language, where we create
+        //    fresh_var = T(input_vars)         and         output_vars ⊆ fresh_var
+        // where the inclusion must be processed to update the languages of output_vars.
+        // Note: It would seem that we could also use this optimization for when we have one input length var (that leads to one automaton). However,
```

Comment:
The splitting and the subsequent substitution will lead to getting stuck in a loop every time, right? It is thus not feasible to add some additional conditions to apply this optimization only for specific shape of such constraints, correct?